In [1]:
print("welcome to my DCF model")
# Financial Inputs for Free Cash Flow Model

print("Enter company financial details:")

# Core operating inputs
ebit = float(input("Enter EBIT: "))
tax_rate = float(input("Enter tax rate (as %, e.g. 25 for 25%): "))

# Capital expenditure
capex = float(input("Enter CapEx (Purchase of PPE): "))

# Depriciation and amortisation
D_A= float(
    input(
        "Enter Depreciation and Amortization (In numbers): "
    )
 )
# Working capital inputs
current_assets_present = float(
    input("Enter Current Assets (Present Year): ")
)
current_assets_previous = float(
    input("Enter Current Assets (Previous Year): ")
)

current_liabilities_present = float(
    input("Enter Current Liabilities (Present Year): ")
)
current_liabilities_previous = float(
    input("Enter Current Liabilities (Previous Year): ")
)


print("\nInputs captured successfully.")

welcome to my DCF model
Enter company financial details:
Enter EBIT: 6837
Enter tax rate (as %, e.g. 25 for 25%): 26
Enter CapEx (Purchase of PPE): 1011.28
Enter Depreciation and Amortization (In numbers): 443
Enter Current Assets (Present Year): 34858
Enter Current Assets (Previous Year): 34063
Enter Current Liabilities (Present Year): 19753
Enter Current Liabilities (Previous Year): 22027

Inputs captured successfully.


welcome to my DCF model

Enter company financial details:

Enter EBIT: 6837

Enter tax rate (as %, e.g. 25 for 25%): 26

Enter CapEx (Purchase of PPE): 1011.28

Enter Depreciation and Amortization (In numbers): 443

Enter Current Assets (Present Year): 34858

Enter Current Assets (Previous Year): 34063

Enter Current Liabilities (Present Year): 19753

Enter Current Liabilities (Previous Year): 22027


In [2]:
#Present year Free Cash Flow

NOPAT=ebit*(1-(tax_rate/100))
Change_WC=(current_assets_present-current_liabilities_present)-(current_assets_previous-current_liabilities_previous)

Free_CF=NOPAT-capex+D_A-Change_WC
print(f"Present Year Free Cash Flow: {Free_CF}")


Present Year Free Cash Flow: 1422.1000000000004


In [3]:
#futute cash flow
# Efficiency metric
reinvested_value=((capex-D_A+Change_WC)/NOPAT)*100
print( f"Reinvested Value rate : {reinvested_value}")

print("For reference ->")
print("Company Type                     ROCE to Use")
print("------------------------------------------------")
print("PSU / mature industrial          18–22%")
print("Strong private industrial        20–25%")
print("Asset-light services             25–30%")
normalised_roce = float(
    input("Enter Normalised ROCE : ")
)
sustainable_GR=(reinvested_value*normalised_roce)/100
print(f"Sustainable Growth Rate : {sustainable_GR}")
print(f"Free Cash flow value on 1st year: {Free_CF}")

Reinvested Value rate : 71.89181283082115
For reference ->
Company Type                     ROCE to Use
------------------------------------------------
PSU / mature industrial          18–22%
Strong private industrial        20–25%
Asset-light services             25–30%
Enter Normalised ROCE : 18
Sustainable Growth Rate : 12.940526309547808
Free Cash flow value on 1st year: 1422.1000000000004


In [7]:
#future cash flow
#free_cf1
#growthrate
g = sustainable_GR/100
# Map t = 0..5 to calendar years
year = 1
years = range(year, year + 6)

# Store results
fcf_values = {}
fcf_list=[]
for t, year in enumerate(years):
    fcf = Free_CF * (1 + g) ** t
    fcf_values[year] = fcf
    fcf_list.append(round(fcf, 2))

# Print results
for year, value in fcf_values.items():
    print(f"Year {year}: FCF = {value:.2f}")

Year 1: FCF = 1422.10
Year 2: FCF = 1606.13
Year 3: FCF = 1813.97
Year 4: FCF = 2048.71
Year 5: FCF = 2313.82
Year 6: FCF = 2613.24


In [8]:
print(fcf_list)

[1422.1, 1606.13, 1813.97, 2048.71, 2313.82, 2613.24]


In [9]:
print("Let's find the WACC")

# Equity & Debt
equity_MC = 306499        # Market Capitalization
debt_BORROW = 61          # Borrowings
interest = 12             # Interest from P&L
taxes = 25                # Tax rate (%)

# CAPM Inputs
rf = 6.6                  # Risk-free rate (%)
rm = 12.6                 # Market return (%)
B = 0.7                   # Beta

# Cost of Debt
cost_of_debt = (interest / debt_BORROW) * 100
print(f"Cost of debt: {cost_of_debt}")

# Cost of Equity (CAPM)
cost_equity = rf + (B * (rm - rf))
print(f"Cost of equity: {cost_equity}")

# Capital Structure
EDE = equity_MC / (equity_MC + debt_BORROW)
DED = debt_BORROW / (equity_MC + debt_BORROW)

print(f"EDE: {EDE}")
print(f"DED: {DED}")

# WACC
WACC = (EDE * cost_equity + DED * cost_of_debt) * (1 - (taxes / 100))

print("WACC---------------->")
print(f"WACC: {WACC}")


Let's find the WACC
Cost of debt: 19.672131147540984
Cost of equity: 10.799999999999999
EDE: 0.9998010177453027
DED: 0.00019898225469728602
WACC---------------->
WACC: 8.10132404749478


In [ ]:
print("Lests find the WACC")
equity_MC=float(input("Enter the equity(market capital): "))
#For cost of debt calculation
print("Enter the below for the calculations of cost of debt")
debt_BORROW=float(input("Enter the debt(borrowings): "))
interest=float(input("Enter the interest(From p&l): "))
cost_of_debt=(interest/debt_BORROW)*100
print(f"Cost of debt: {cost_of_debt}")
taxes=float(input("Enter the tax rate(from P&L): "))
# For the calculation of cost of equity
print("Rf = risk free rate of 10years of gov bonds so")
print("So its better to keep it same in most cases")
Rf=("6-7")
print("So its better to keep it same in most cases")
rf=float(input(f"Enter number of years (recommended {Rf}): "))
print("Rm- expected return of the entire market")
Rm=("12-13")
rm=float(input(f"Enter number of years (recommended {Rm}): ")) # Converted rm to float
print("beta(B)=“If the market moves by 1%, how much does this stock’s return move?”")
print("β = 1 → stock moves same as market")
print("β > 1 → more volatile than market (higher risk)")
print("β < 1 → less volatile than market (lower risk)")
print("β < 0 → moves opposite to market (rare)")
B=float(input("Enter the beta(check on comapny financials): "))
cost_equity= rf+(B*(rm-rf))
print(f"Cost of equity: {cost_equity}")
EDE=equity_MC/(equity_MC+debt_BORROW)
DED=debt_BORROW/(equity_MC+debt_BORROW)
print(f"EDE: {EDE}")
print(f"DED: {DED}")
WACC=((EDE)*cost_equity+(DED)*cost_of_debt)*(1-(taxes/100))
print("WACC---------------->")
print(f"WACC: {WACC}")

Lests find the WACC
Enter the equity(market capital): 306499
Enter the below for the calculations of cost of debt
Enter the debt(borrowings): 61
Enter the interest(From p&l): 12
Cost of debt: 19.672131147540984
Enter the tax rate(from P&L): 25
Rf = risk free rate of 10years of gov bonds so
So its better to keep it same in most cases
So its better to keep it same in most cases
Enter number of years (recommended 6-7): 6.6
Rm- expected return of the entire market
Enter number of years (recommended 12-13): 12.6
beta(B)=“If the market moves by 1%, how much does this stock’s return move?”
β = 1 → stock moves same as market
β > 1 → more volatile than market (higher risk)
β < 1 → less volatile than market (lower risk)
β < 0 → moves opposite to market (rare)
Enter the beta(check on comapny financials): 0.7
Cost of equity: 10.799999999999999
EDE: 0.9998010177453027
DED: 0.00019898225469728602
WACC---------------->
WACC: 8.10132404749478


In [11]:
pv_fcf_list=[]
for i in fcf_list:
    pv_fcf=i/((1+WACC/100)**(fcf_list.index(i)+1))
    pv_fcf_list.append(round(pv_fcf,2))
print(pv_fcf_list)

for i in range(len(pv_fcf_list)):
    print(f"Year {i+1}: PV FCF = {pv_fcf_list[i]}")

[1315.53, 1374.42, 1435.94, 1500.23, 1567.38, 1637.54]
Year 1: PV FCF = 1315.53
Year 2: PV FCF = 1374.42
Year 3: PV FCF = 1435.94
Year 4: PV FCF = 1500.23
Year 5: PV FCF = 1567.38
Year 6: PV FCF = 1637.54


In [ ]:
#This is for the calculation of terminal value
print("growth rate(G)-one constant long-term growth rate applied after the forecasted year")
print("Conservative:                   3% – 4%")
print("Reasonable / standard:          4% – 5%")
print("Aggressive:                     5% – 6%")
G=float(input("Enter the growth rate : "))

TV=fcf_list[-2]/((WACC-G)/100)
print(f"Terminal Value: {TV}")
#Present value of Terminal Value
N=input("Enter the number of years: ")
PV_TV=TV/((1+WACC/100)**int(N))
print(f"Present Value of Terminal Value: {PV_TV}")


growth rate(G)-one constant long-term growth rate applied after the forecasted year
Conservative:                   3% – 4%
Reasonable / standard:          4% – 5%
Aggressive:                     5% – 6%
Enter the growth rate : 3.5
Terminal Value: 50285.93688390506
Enter the number of years: 5
Present Value of Terminal Value: 34063.673339740875
